darshan-util/darshan-mpiio-logutils.c

```diff
@@ -518,42 +585,62 @@ static void darshan_log_agg_mpiio_files(void *rec, void *agg_rec, int init_flag)
-                    agg_mpi_rec->fcounters[MPIIO_F_FASTEST_RANK_TIME] = mpi_time;
+                    agg_mpi_rec->counters[MPIIO_FASTEST_RANK] = -1;
+                    agg_mpi_rec->counters[MPIIO_FASTEST_RANK_BYTES] = -1;
+                    agg_mpi_rec->fcounters[MPIIO_F_FASTEST_RANK_TIME] = 0.0;
+                    break;
                 }
-
-                if(mpi_time < agg_mpi_rec->fcounters[MPIIO_F_FASTEST_RANK_TIME])
-                {
-                    agg_mpi_rec->counters[MPIIO_FASTEST_RANK] = mpi_rec->base_rec.rank;
-                    agg_mpi_rec->counters[MPIIO_FASTEST_RANK_BYTES] = mpi_bytes;
-                    agg_mpi_rec->fcounters[MPIIO_F_FASTEST_RANK_TIME] = mpi_time;
+                if (init_flag ||
+                    mpi_fastest_time < agg_mpi_rec->fcounters[MPIIO_F_FASTEST_RANK_TIME]) {
+                    /* The incoming record wins if a) this is the first
+                     * record we are aggregating or b) it is the fastest
+                     * record we have seen so far.
+                     */
+                    agg_mpi_rec->counters[MPIIO_FASTEST_RANK]
+                        = mpi_fastest_rank;
+                    agg_mpi_rec->counters[MPIIO_FASTEST_RANK_BYTES]
+                        = mpi_fastest_bytes;
+                    agg_mpi_rec->fcounters[MPIIO_F_FASTEST_RANK_TIME]
+                        = mpi_fastest_time;
                 }
                 break;
             case MPIIO_F_SLOWEST_RANK_TIME:
-                if(init_flag)
+                if(!shared_file_flag)
                 {
-                    /* set slowest rank counters according to root rank. these counters
-                     * will be determined as the aggregation progresses.
+                    /* The slowest counters are only valid under these
+                     * conditions when aggregating records that all refer to
+                     * the same file.
                      */
-                    agg_mpi_rec->counters[MPIIO_SLOWEST_RANK] = mpi_rec->base_rec.rank;
-                    agg_mpi_rec->counters[MPIIO_SLOWEST_RANK_BYTES] = mpi_bytes;
-                    agg_mpi_rec->fcounters[MPIIO_F_SLOWEST_RANK_TIME] = mpi_time;
+                    agg_mpi_rec->counters[MPIIO_SLOWEST_RANK] = -1;
+                    agg_mpi_rec->counters[MPIIO_SLOWEST_RANK_BYTES] = -1;
+                    agg_mpi_rec->fcounters[MPIIO_F_SLOWEST_RANK_TIME] = 0.0;
+                    break;
                 }
-
-                if(mpi_time > agg_mpi_rec->fcounters[MPIIO_F_SLOWEST_RANK_TIME])
-                {
-                    agg_mpi_rec->counters[MPIIO_SLOWEST_RANK] = mpi_rec->base_rec.rank;
-                    agg_mpi_rec->counters[MPIIO_SLOWEST_RANK_BYTES] = mpi_bytes;
-                    agg_mpi_rec->fcounters[MPIIO_F_SLOWEST_RANK_TIME] = mpi_time;
+                if (init_flag ||
+                    mpi_slowest_time < agg_mpi_rec->fcounters[MPIIO_F_SLOWEST_RANK_TIME]) {
```

Comment:
Looks like we need to flip the the `<` to `>` here (and in POSIX and STDIO codepaths). For slowest time, we want the highest number.

Reply:
Thanks @shanedsnyder , you are right!  I just pushed a fix.